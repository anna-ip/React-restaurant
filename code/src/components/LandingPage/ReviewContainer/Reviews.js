import React, { useContext, useState, useEffect } from 'react'
import { ReviewContext } from '../../../App'
import Card from './Card'
import Arrows from './Arrows'
import styles from './Reviews.module.scss'

const Reviews = () => {
  const [storeReviews] = useContext(ReviewContext)
  const reviews = storeReviews.reviews

  const [reviewData, setReviewData] = useState(reviews)
  const [current, setCurrent] = useState(0)
  const [quote, getQuote] = useState(reviewData[current])

  useEffect(() => {
    getQuote(reviewData[current])
  }, [current, reviewData, quote])

  const nextReview = () => {
    current === reviewData.length - 1 ? setCurrent(0) : setCurrent(current + 1)
  }

  const prevRewiev = () => {
    current === 0 ? setCurrent(reviewData.length - 1) : setCurrent(current - 1)
  }

  return (
    <div className={styles.container}>
      <h3>Reviews</h3>
      <section>
        <span className={styles.coutes}>,,</span>
        <Card quote={quote} />
        <Arrows nextReview={nextReview} prevRewiev={prevRewiev} />
      </section>
    </div>
  )
}

export default Reviews
