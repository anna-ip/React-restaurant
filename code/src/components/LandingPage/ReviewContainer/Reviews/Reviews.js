import React, { useState, useEffect, useContext } from 'react'
import { ReviewContext } from '../../../../App'
import Card from '../Card/Card'
import Arrows from '../Arrows/Arrows'
import styles from './Reviews.module.scss'

const Reviews = () => {
  const [storeReviews] = useContext(ReviewContext)
  const reviews = storeReviews.reviews
  //console.log(JSON.stringify(reviews))

  const [current, setCurrent] = useState(0)
  console.log('current', current)
  const [quote, getQuote] = useState(reviews[current])
  console.log('quote', quote)

  useEffect(() => {
    getQuote(reviews[current])
  }, [current, reviews, quote])

  const nextReview = () => {
    current === reviews.length - 1 ? setCurrent(0) : setCurrent(current + 1)
  }

  const prevRewiev = () => {
    current === 0 ? setCurrent(reviews.length - 1) : setCurrent(current - 1)
  }

  return (
    <div className={styles.container}>
      <h3>Reviews</h3>
      <section>
        <span className={styles.qoutes}>,,</span>
        <Card quote={quote} />
        <Arrows nextReview={nextReview} prevRewiev={prevRewiev} />
      </section>
    </div>
  )
}

export default Reviews
