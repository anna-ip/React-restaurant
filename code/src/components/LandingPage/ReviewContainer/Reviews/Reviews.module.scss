@import '../../../../styles/_variables.scss';
@import '../../../../styles/_mixins.scss';

.container {
  position: relative;
  z-index: 1;
  width: 100%;
  height: 360px;
  margin: 3rem 0 3rem 0;
  background-color: $turquise-color;

  @media (min-width: $layout-breakpoint-small) {
    width: 90%;
    height: 300px;
  }

  h3 {
    font-family: $second-font-family;
    color: $dark-color;
    font-size: 2rem;
    font-weight: 400;
    margin: 1.5rem 0 0 0;
  }
  section {
    @include flexCenter;
  }
}
.qoutes {
  color: #fff;
  font-size: 2.2rem;
  height: 2rem;
}
