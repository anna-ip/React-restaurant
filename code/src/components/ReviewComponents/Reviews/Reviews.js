import React, { useContext } from 'react'
import { ReviewContext } from '../../../App'
import Card from '../Card/Card'
import Arrows from '../Arrows/Arrows'
import styles from './Reviews.module.scss'
import useCarousel from '../../../hooks/useCarousel'

const Reviews = () => {
  const [storeReviews] = useContext(ReviewContext)
  const reviews = storeReviews.reviews

  const [nextReview, prevReview, quote] = useCarousel(reviews)

  return (
    <div className={styles.container}>
      <h3>Reviews</h3>
      <section>
        <span className={styles.qoutes}>,,</span>
        <Card quote={quote} />
        <Arrows nextReview={nextReview} prevReview={prevReview} />
      </section>
    </div>
  )
}

export default Reviews
