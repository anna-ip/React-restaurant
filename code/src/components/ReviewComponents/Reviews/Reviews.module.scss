@import '../../../styles/_variables.scss';
@import '../../../styles/_mixins.scss';

.container {
  position: relative;
  z-index: 1;
  width: 100%;
  height: 27rem;
  margin: 3rem 0 3rem 0;
  background-color: $turquise-color;

  @media (min-width: $layout-breakpoint-small) {
    width: 90%;
  }

  h3 {
    font-family: $second-font-family;
    color: $dark-color;
    font-size: 2rem;
    font-weight: 400;
    margin: 1.5rem 0 0 0;
  }
  section {
    @include flexCenter;
  }

  .link {
    position: absolute;
    bottom: 3%;
  }
}

