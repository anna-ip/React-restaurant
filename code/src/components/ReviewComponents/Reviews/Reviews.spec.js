import React from "react";
import Reviews from "./Reviews";
import { shallow } from "enzyme";

jest.mock("React", () => ({
  ...jest.requireActual("React"),
  useContext: () => [fakeContext],
}));

const fakeContext = {
  reviews: [
    {
      id: 1,
      date: "Mon May 31 2021",
      name: "susan smith",
      text: "I'm baby meggings twee health goth +1. Bicycle rights tumeric chartreuse before they sold out chambray pop-up. Shaman humblebrag pickled coloring book salvia hoodie, cold-pressed four dollar toast everyday carry",
    },
    {
      id: 2,
      date: "Mon May 31 2021",
      name: "anna johnson",
      text: "Helvetica artisan kinfolk thundercats lumbersexual blue bottle. Disrupt glossier gastropub deep v vice franzen hell of brooklyn twee enamel pin fashion axe.photo booth jean shorts artisan narwhal.",
    },
    {
      id: 3,
      date: "Mon May 31 2021",
      name: "peter jones",
      text: "Sriracha literally flexitarian irony, vape marfa unicorn. Glossier tattooed 8-bit, fixie waistcoat offal activated charcoal slow-carb marfa hell of pabst raclette post-ironic jianbing swag.",
    },
    {
      id: 4,
      date: "Mon May 31 2021",
      name: "bill anderson",
      text: "Edison bulb put a bird on it humblebrag, marfa pok pok heirloom fashion axe cray stumptown venmo actually seitan. VHS farm-to-table schlitz, edison bulb pop-up 3 wolf moon tote bag street art shabby chic.",
    },
  ],
};

afterEach(() => {
  jest.clearAllMocks();
});

describe("<Reviews/>", () => {
  test("<Renders correctly using enzyme", () => {
    const wrapper = shallow(<Reviews />);
    expect(wrapper).toMatchSnapshot();
  });

  test("reviews exist", () => {
    const wrapper = shallow(<Reviews />);
    expect(wrapper.exists()).toBeTruthy();
  });
});
