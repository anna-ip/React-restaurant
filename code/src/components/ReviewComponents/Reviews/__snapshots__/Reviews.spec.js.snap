// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Reviews/> <Renders correctly using enzyme 1`] = `
<article
  className="container"
>
  <h3>
    Reviews
  </h3>
  <section>
    <span
      className="qoutes"
    >
      ,,
    </span>
    <Card
      quote={
        Object {
          "id": 1,
          "name": "susan smith",
          "text": "I'm baby meggings twee health goth +1. Bicycle rights tumeric chartreuse before they sold out chambray pop-up. Shaman humblebrag pickled coloring book salvia hoodie, cold-pressed four dollar toast everyday carry",
        }
      }
    />
    <Arrows
      nextReview={[Function]}
      prevReview={[Function]}
    />
  </section>
</article>
`;
